### PR TITLE
Improved FAQ

### PR DESCRIPTION
**Pull request by adaedra**
- Rewritted most entries of Generic and Setup
- Filled missing entries in Syntax
- Enabled HTML rendering for FAQ answers

### CODE REVIEW

File: db/seeds/all/faq.yaml [hunk 1]
```diff
@@ -1,3 +1,58 @@
...
+      It all depends on a large number of parameters.
+
+  - - Should I use Ruby or X?
+    - It depends on many parameters. Ruby is a general purpose language which can achieve
+      anything. Then, deciding on if you should use it or not is all up to you, but
+      know that there is no good answer to this question.
+
+  - - Is Ruby slow?
+    - Ruby is an interpreted language, so it will be slower than native languages as
+      C, but will be close to python or perl. Ruby is fast to write, it can save you
```
Comment (jhass) [inline db/seeds/all/faq.yaml]: Nitpick: "Python", "Perl".

File: db/seeds/all/faq.yaml [hunk 2]
```diff
@@ -1,3 +1,58 @@
...
+  - - Will Ruby die?
+    - Ruby is around for a long time, well over 20 years already, major companies depend
+      on it, it is actively developed and available in the repositories of every major
+      distribution. No, Ruby will not go away anytime soon.
+
+  - - 'What''s the difference between this channel and #ruby-lang?'
+    - "#ruby is larger than #ruby-lang, #ruby-lang is officially linked to by ruby-lang.org"
+
+  - - In Rails, how can I …?
+    - "#ruby is targeted at general ruby discussion. If your question targets Ruby on
```
Comment (jhass) [inline db/seeds/all/faq.yaml]: Nitpick: "Ruby" (yeah, I wrote this, please save my back :P)

File: db/seeds/all/faq.yaml [hunk 3]
```diff
@@ -25,6 +80,18 @@ Syntax:
...
+  - - How can I pass a method as an argument to another method?  
+    - |
+      You can get a callable object from a method by calling <code>#method</code>
+      on the object, for example <code>"hello".method(:length)</code>. You obtain
+      a <code>method</code> object which behaves like a <code>Proc</code> which you
+      can pass to other functions. 
+
+  - - Why does `"b" == "a" || "b"` not return true?  
+    - |
+      The <code>||</code> operator returns the argument which equals to true. In this
```
Comment (jhass) [inline db/seeds/all/faq.yaml]: "which is truthy" might be better here. Do we already have an entry for truthiness and falseiness we can reference?

File: db/seeds/all/faq.yaml [hunk 4]
```diff
@@ -25,6 +80,18 @@ Syntax:
...
+      case, <code>"b" == "a"</code> is false, but "b" equals to true, so "b" is returned.
```
Comment (jhass) [inline db/seeds/all/faq.yaml]: "returns false" or "evaluates to false" might be better here.

File: db/seeds/all/faq.yaml [hunk 5]
```diff
@@ -1,3 +1,58 @@
...
+      IDE.
+
+Setup:
+  - - How do I install Ruby?
+    - |-
+      <p>There are multiple ways of getting ruby.</p>
+      <ul>
+      <li>On Linux, use your system package manager.</li>
+      <li>On Windows, you can use <a href="http://rubyinstaller.org/">Ruby installer</a>.</li>
+      <li>OS X comes with a pre-installed ruby version.</li>
```
Comment (jhass) [inline db/seeds/all/faq.yaml]: "Ruby" :)

File: db/seeds/all/faq.yaml (hunk 3 above)
Comment (adaedra) [inline db/seeds/all/faq.yaml]: It seems not. Maybe write one, or refer to another online documentation on the subject, as it is rather large?

Comment (jhass) [inline db/seeds/all/faq.yaml]: Well, is it? `nil` and `false` are the only objects that are considered falsey in Ruby, everything else is truthy. Besides that we only need to mention that this applies to conditions in control flow statements like `if` and in filtering methods such as `Enumerable#select`.

Comment (adaedra) [inline db/seeds/all/faq.yaml]: Presented like that, it can have it's own FAQ section. But under what kind of question? "What are booleans?"

Comment (jhass) [inline db/seeds/all/faq.yaml]: "Why does `if 0` pass?" perhaps? I think that's the most common pitfall, coming from C or PHP at least.

**Comment by apeiros** [comment]
Thank you @adaedra and @jhass for all the work!

**Pull request merged**